From 87e02e93935d2dcc7d48da6ddeed4eb893701610 Mon Sep 17 00:00:00 2001
From: Gabor Abonyi <gabor.abonyi@arm.com>
Date: Wed, 31 Jan 2024 11:56:52 +0100
Subject: [PATCH 4/4] Move activation_buf_dram to .bss.NoInit region

This is needed to place activation_buf_dram in a no init region, so
it does not require space in flash.

Signed-off-by: Gabor Abonyi <gabor.abonyi@arm.com>
---
 source/hal/source/components/npu/include/ethosu_mem_config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/hal/source/components/npu/include/ethosu_mem_config.h b/source/hal/source/components/npu/include/ethosu_mem_config.h
index 2cec141..f9ec205 100644
--- a/source/hal/source/components/npu/include/ethosu_mem_config.h
+++ b/source/hal/source/components/npu/include/ethosu_mem_config.h
@@ -49,7 +49,7 @@
     #define ACTIVATION_BUF_SECTION      section(".bss.NoInit.activation_buf_sram")
     #define ACTIVATION_BUF_SECTION_NAME ("SRAM")
 #elif (ETHOS_U_NPU_MEMORY_MODE==ETHOS_U_NPU_MEMORY_MODE_DEDICATED_SRAM)
-    #define ACTIVATION_BUF_SECTION      section("activation_buf_dram")
+    #define ACTIVATION_BUF_SECTION      section(".bss.NoInit.activation_buf_dram")
     #define CACHE_BUF_SECTION           section(".bss.NoInit.ethos_u_cache")
     #define ACTIVATION_BUF_SECTION_NAME ("DDR/DRAM")
     #define CACHE_BUF_ATTRIBUTE         __attribute__((aligned(ETHOS_U_MEM_BYTE_ALIGNMENT), CACHE_BUF_SECTION))
-- 
2.40.1

